lib/asyn-thread.c

```diff
@@ -454,6 +628,34 @@ static bool init_resolve_thread(struct Curl_easy *data,
   /* The thread will set this to 1 when complete. */
   td->tsd.done = 0;
 
+#ifdef _WIN32
+  if(Curl_isWindows8OrGreater && ptrFreeAddrInfoExW &&
+      ptrGetAddrInfoExCancel && ptrGetAddrInfoExW) {
+    int hostname_w_len = MultiByteToWideChar(CP_UTF8, MB_ERR_INVALID_CHARS,
+        hostname, -1, NULL, 0);
+    int name_sz = sizeof(td->tsd.w8.name) / sizeof(td->tsd.w8.name[0]);
+    if(hostname_w_len > 0 && hostname_w_len < name_sz)
+      hostname_w_len = MultiByteToWideChar(CP_UTF8, 0, hostname, -1,
+        td->tsd.w8.name, hostname_w_len);
+    if(hostname_w_len > 0 && hostname_w_len < name_sz) {
+      swprintf(td->tsd.w8.port,
+        sizeof(td->tsd.w8.port) / sizeof(td->tsd.w8.port[0]),
+        L"%d", port);
+      td->tsd.w8.hints.ai_family = hints->ai_family;
+      td->tsd.w8.hints.ai_socktype = hints->ai_socktype;
+      td->complete_ev = CreateEvent(NULL, TRUE, FALSE, NULL);
```

Comment:
if `CreateEvent` returns NULL here, is that not an out of memory error or similar that should rather cause an error to be returned?

Reply:
`td->complete_ev != NULL` indicates that GetAddrInfoExW code is to be used instead of manual thread. If it failed there, then it simply falls back to using regular thread code.

Reply:
updated to error out